File: gestaolegal/models/orientacao_juridica.py
from dataclasses import dataclass
from datetime import datetime
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from gestaolegal.models.atendido import Atendido
    from gestaolegal.models.user import UserInfo


@dataclass
class OrientacaoJuridica:
    area_direito: str
    sub_area: str | None
    descricao: str
    status: int
    data_criacao: datetime
    id: int | None = None
    id_usuario: int | None = None

    atendidos: list["Atendido"] | None = None
    usuario: "UserInfo | None" = None


@dataclass
class OrientacaoJuridicaDetail:
    id: int
    area_direito: str
    sub_area: str | None
    data_criacao: datetime
    status: bool
    atendidos: list["Atendido"]
    descricao: str
    usuario: "UserInfo | None"


@dataclass
class OrientacaoJuridicaListItem:
    id: int
    area_direito: str
    sub_area: str | None
    descricao: str
    atendidos: list[str]
    usuario: "UserInfo | None"
    data_criacao: datetime
    status: bool

File: web/src/lib/types/orientacao-juridica.ts
export type OrientacaoJuridica = {
	id: number;
	area_direito: string;
	sub_area?: string | null;
	descricao: string;
	data_criacao?: string | null;
	status: boolean;
	usuario?: {
		id: number;
		nome: string;
	} | null;
	atendidos?: Array<{
		id: number;
		nome: string;
	}>;
};

export type PaginatedOrientacoesJuridicas = {
	items: OrientacaoJuridica[];
	total: number;
	page: number;
	pages: number;
	per_page: number;
};
